phpBB/install/startup.php

```diff
@@ -227,6 +227,7 @@ function installer_shutdown_function($display_errors)
 
 // Include files
 phpbb_require_updated('includes/compatibility_globals.' . $phpEx, $phpbb_root_path);
+phpbb_require_updated('includes/constants.' . $phpEx, $phpbb_root_path);
```

Comment:
I guess it should be already done here https://github.com/phpbb/phpbb/blob/master/phpBB/phpbb/install/helper/container_factory.php#L187

Reply:
Yes, I have some errors doing that, that is why i have set wip. But withtout that there is an error during install when you have error_reporting(-1)

`
    phpbb: [phpBB Debug] "Use of undefined constant ANONYMOUS - assumed 'ANONYMOUS' (this will throw an Error in a future version of PHP)" in file /home/vagrant/phpbb/phpBB/phpbb/install/console/command/install/install.php on line 69
    phpbb: [phpBB Debug] "Use of undefined constant ANONYMOUS - assumed 'ANONYMOUS' (this will throw an Error in a future version of PHP)" in file /home/vagrant/phpbb/phpBB/phpbb/install/console/command/install/install.php on line 69
    phpbb: 
    phpbb:       0/13  ░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░    0%
`

Reply:
Ignore the last commit for now

Reply:
@rubencm probably you have the source code outdated a bit. The bug was fixed with this commit https://github.com/phpbb/phpbb/commit/09e308a06d7fcaa624afe7e4a86e178bab84a362#diff-8bfd43635044b97335b9c309cd081a1a

Reply:
No, i have that updated

Reply:
In fact, you can see that the error is the same, but not in the same file.

Reply:
Indeed. That is because of `\phpbb\user` initialization in the command constructor without `ANONYMOUS` constant being defined.
Okay.

Reply:
If you want to look at it, it happens during installation with cli. I will remove this commit.

Reply:
the real issue for both cases is this line https://github.com/phpbb/phpbb/blob/1e40c814f39c986de6c0e67723026beb4f078e34/phpBB/phpbb/session.php#L30

I couldn't find any use cases of this property excepting a couple of assignments in session.php https://github.com/phpbb/phpbb/blob/1e40c814f39c986de6c0e67723026beb4f078e34/phpBB/phpbb/session.php#L368 and https://github.com/phpbb/phpbb/blob/1e40c814f39c986de6c0e67723026beb4f078e34/phpBB/phpbb/session.php#L655
but it looks like `->id` is not used anywhere. Probably just drop it in 4.0?

Reply:
Hmm. It was added by @mrgoldy not a long ago here https://github.com/phpbb/phpbb/commit/e5c80e46b09b90ec2f5e181032c66480af598e51#diff-ef9a28b1448b3c0a926f41845507b78fR28